base/gmp.jl

```diff
@@ -458,7 +475,7 @@ powermod(x::Integer, p::Integer, m::BigInt) = powermod(big(x), big(p), m)
 
 function gcdx(a::BigInt, b::BigInt)
     if b == 0 # shortcut this to ensure consistent results with gcdx(a,b)
-        return a < 0 ? (-a,-one(BigInt),zero(BigInt)) : (a,one(BigInt),zero(BigInt))
+        return a < 0 ? (-a,-ONE,ZERO) : (a,ONE,ZERO)
```

Comment:
I don't think it's OK. I don't know why you think https://github.com/JuliaLang/julia/pull/15546 is supporting this change here. In https://github.com/JuliaLang/julia/pull/15546 no decision was made about the whole aliasing thing, but people were complaining about undefined aliasing behavior, that is, runtime dependent aliasing behavior. Here you introduce yet another unnecessary suprising aliasing behavior, which will yield to very subtle bugs.

If julia wraps GMP's `mpz` as `BigInt`, then it should be possible to call any function from GMP with `BigInt`s. With this change this not safe anymore. You would always have to look up, if the wrap julia function is returning the newly introduced global variables. And I am not talking about public API, but also about future modifications of the GMP module.

```
a,b,c = gcd(BigInt(1), BigInt(0))
set_zero!(b)
```
This code snipped should never change the behavior of julia, but with the proposed change it will. It leads to runtime changes in all the functions call `ONE` and `ZERO`.